internal/bind/params.go

```diff
@@ -177,63 +244,82 @@ func toYdbParam(name string, value interface{}) (table.ParameterOption, error) {
 	params = make([]table.ParameterOption, 0, len(args))
 	for i, arg := range args {
+		var err error
 		switch x := arg.(type) {
 		case driver.NamedValue:
-			if x.Name == "" {
-				switch xx := x.Value.(type) {
-				case *table.QueryParameters:
-					if len(args) > 1 {
-						return nil, xerrors.WithStackTrace(errMultipleQueryParameters)
-					}
-					xx.Each(func(name string, v types.Value) {
-						params = append(params, table.ValueParam(name, v))
-					})
-				case table.ParameterOption:
-					params = append(params, xx)
-				default:
-					x.Name = fmt.Sprintf("$p%d", i)
-					param, err := toYdbParam(x.Name, x.Value)
-					if err != nil {
-						return nil, xerrors.WithStackTrace(err)
-					}
-					params = append(params, param)
-				}
-			} else {
-				param, err := toYdbParam(x.Name, x.Value)
-				if err != nil {
-					return nil, xerrors.WithStackTrace(err)
-				}
-				params = append(params, param)
-			}
+			params, err = processNamedValue(params, x, i, len(args))
 		case sql.NamedArg:
-			if x.Name == "" {
-				return nil, xerrors.WithStackTrace(errUnnamedParam)
-			}
-			param, err := toYdbParam(x.Name, x.Value)
-			if err != nil {
-				return nil, xerrors.WithStackTrace(err)
-			}
-			params = append(params, param)
+			params, err = appendParam(params, x.Name, x.Value)
 		case *table.QueryParameters:
 			if len(args) > 1 {
-				return nil, xerrors.WithStackTrace(errMultipleQueryParameters)
+				return nil, errWithStackTrace(errMultipleQueryParameters)
 			}
 			x.Each(func(name string, v types.Value) {
 				params = append(params, table.ValueParam(name, v))
 			})
 		case table.ParameterOption:
 			params = append(params, x)
 		default:
-			param, err := toYdbParam(fmt.Sprintf("$p%d", i), x)
-			if err != nil {
-				return nil, xerrors.WithStackTrace(err)
-			}
-			params = append(params, param)
+			params, err = appendParam(params, fmt.Sprintf("$p%d", i), x)
+		}
+		if err != nil {
+			return nil, err
 		}
 	}
+	sortParamsByName(params)
+
+	return params, nil
+}
+
+func processNamedValue(
+	params []table.ParameterOption,
+	x driver.NamedValue,
+	i, argsLen int,
+) ([]table.ParameterOption, error) {
+	if x.Name == "" {
+		return processUnnamedNamedValue(params, x, i, argsLen)
+	}
+
+	return appendParam(params, x.Name, x.Value)
+}
+
+func processUnnamedNamedValue(
+	params []table.ParameterOption,
+	x driver.NamedValue,
+	i, argsLen int,
+) ([]table.ParameterOption, error) {
+	switch xx := x.Value.(type) {
+	case *table.QueryParameters:
+		if argsLen > 1 {
+			return nil, errWithStackTrace(errMultipleQueryParameters)
+		}
+		xx.Each(func(name string, v types.Value) {
+			params = append(params, table.ValueParam(name, v))
+		})
+
+		return params, nil
+	case table.ParameterOption:
+		return append(params, xx), nil
+	default:
+		return appendParam(params, fmt.Sprintf("$p%d", i), x.Value)
+	}
+}
+
+func appendParam(params []table.ParameterOption, name string, value interface{}) ([]table.ParameterOption, error) {
```

Comment:
it isn't append parameter only, it additionally convert param to table.ParameterOption (`*params.Parameter` in main branch).